lib/absinthe/phase/error.ex

```diff
@@ -15,18 +16,21 @@ defmodule Absinthe.Phase.Error do
     message: String.t,
     phase: module,
     locations: [loc_t],
+    extra: Keyword.t
   }
 
   @doc """
   Generate a phase error that relates to a specific point in the
   document.
   """
   @spec new(Absinthe.Phase.t, String.t, loc_t | [loc_t]) :: t
-  def new(phase, message, location) do
+  @spec new(Absinthe.Phase.t, String.t, loc_t | [loc_t], Keyword.t) :: t
+  def new(phase, message, location, extra \\ []) do
     %__MODULE__{
       phase: phase,
       message: message,
-      locations: List.wrap(location)
+      locations: List.wrap(location),
+      extra: Enum.filter(extra, &filter_extra_item/1)
```

Comment:
Also, I'm not excited about the prospect of putting everything that's additional in `extra` -- is this a convention from somewhere else? The spec just states that "GraphQL servers may provide additional entries to error as they choose to produce more helpful or machine‐readable errors, however future versions of the spec may describe additional entries to errors."

I'd argue that if I give `{:error, message: "Bad", something: "Else"}`, I should get the `{"message": "Bad", "something": "Else", "locations": [...]}` as the JSON error -- no need to sandbox the additional stuff under `"extras"` -- if people want to do that, they can use, eg, `{:error, message: "Bad", extras: [something: "Else"]}` themselves.